demos/app/iot_demo_arguments.c

```diff
@@ -133,22 +143,37 @@ static bool _validateArguments( const IotDemoArguments_t * pArguments )
             IOT_SET_AND_GOTO_CLEANUP( false );
         }
 
-        /* Check that a client certificate path was set. */
-        if( ( pArguments->pClientCertPath == NULL ) ||
-            ( strlen( pArguments->pClientCertPath ) == 0 ) )
+        /* If the host is connecting to the MQTT broker hosted by AWS IoT Core,
+         * there must either be a set of X.509 credentials or a
+         * username/password. Therefore, check that here in order to facilitate
+         * debugging. For other MQTT brokers, assume that the CLI arguments are
+         * as intended.
+         */
+        if( pArguments->awsIotMqttMode == true )
```

Comment:
Why are we not checking for the validity of the non AWS IoT MQTT Broker arguments?

Reply:
Please see my comment below. Because of that, in the non-AWS case, we don't know what the server-side policies will be with respect to web server and MQTT broker authentication, so we can't be as specific in our CLI argument checks.